lib/compat/wordpress-6.3/get-global-styles-and-settings.php

```diff
@@ -158,3 +158,38 @@ function gutenberg_get_global_styles( $path = array(), $context = array() ) {
 
 	return _wp_array_get( $styles, $path, $styles );
 }
+
+/**
+ * Equivalent to `wp_get_global_styles` but replaces CSS variables with their values.
+ *
+ * @since 6.3.0
+ *
+ * @param array $path    Path to the specific style to retrieve. Optional.
+ *                       If empty, will return all styles.
+ * @param array $context {
+ *     Metadata to know where to retrieve the $path from. Optional.
+ *
+ *     @type string $block_name Which block to retrieve the styles from.
+ *                              If empty, it'll return the styles for the global context.
+ *     @type string $origin     Which origin to take data from.
+ *                              Valid values are 'all' (core, theme, and user) or 'base' (core and theme).
+ *                              If empty or unknown, 'all' is used.
+ * }
+ * @see wp_get_global_styles()
+ * @return array The styles to retrieve with their values.
+ */
+function gutenberg_get_global_style_values( $path = array(), $context = array() ) {
```

Comment:
How do you envision this function to be used with the other one (resolving the references)? In other words, what's the API to compose these transformations?

Through the conversation we've had, I see two options forward:

- **Standalone functions**. We'd have `gutenberg_transform_style_from_variable_to_value` and `gutenberg_transform_style_from_reference_to_value`, respectively. We can compose them:

```php
gutenberg_get_global_styles(); // Some consumers don't want those transformations.
gutenberg_transform_style_from_variable_to_value( gutenberg_get_global_styles() ); // Some do not want vars.
gutenberg_transform_style_from_reference_to_value( gutenberg_get_global_styles() ); // Some do not want refs.
gutenberg_transform_style_from_variable_to_value( gutenberg_transform_style_from_reference_to_value( gutenberg_get_global_styles() ) ); // Some do not want either.
```

The new functions should be idempotent, meaning that `get styles > transform from variables > transform from references` should yield the same result as `get styles > transform from references > transform from variables`.

- **Parameter to the existing function**. Either expanding the existing `$context` or a new `$transforms` parameter. Not super opinionated about it.

```php
gutenberg_get_global_styles(); // Some consumers don't want the transformations.
gutenberg_get_global_styles( array(), array( 'transforms' => array( 'resolve-variables' ) ) ); // Some do not want vars.
gutenberg_get_global_styles( array(), array( 'transforms' => array( 'resolve-references' ) ) ); // Some do not want refs.
gutenberg_get_global_styles( array(), array( 'transforms' => array( 'resolve-references', 'resolve-variables' ) ) ); // Some do not want either.
```

I'm leaning towards the parameter approach.

Reply:
Pinging some people that had been involved in these conversations at https://github.com/WordPress/gutenberg/issues/49712 https://github.com/WordPress/gutenberg/issues/49715 and https://github.com/WordPress/gutenberg/issues/45171

cc @jsnajdr @ramonjd @geriux @Aljullu @jeherve @BogdanUngureanu 

Is there any other alternative? What are your thoughts?

Reply:
I'm more attracted to the idea of controlling the output via parameters for the reasons you mentioned — some users might want ref resolution only etc.

The function signature already features two arguments: `gutenberg_get_global_styles( $path = array(), $context = array() ) ` — are we thinking of loading the `$context` with transform options? I think that sounds reasonable, just making sure ;) Backwards compat should be unaffected.

Looking at the code in this PR and even the [messy function I wrote to resolve node refs](https://github.com/WordPress/gutenberg/pull/49737/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R3049), it doesn't look like it will bloat the function too much.

Reply:
@oandregal Initially I was more leaned towards a new function, but after adding that and figuring that technically the function parameters stay the same, I'm also more inclined towards a parameter approach, and thanks to your suggested example, I see more clearly that this approach is more viable.